lib/kdtree.h

```diff
@@ -66,6 +67,9 @@ class FlatNode {
     constexpr static uint32_t INVALID_TRIANGLE_ID = 0xFFFFFFFF >> 2;
 
 public:
+
+    FlatNode() {};
```

Comment:
I think this is not a big deal. It probably has to initialise vector space somehow, without a default ctor this is not that easy.

Lets replace it by:
`FlatNode() = default`.